### PR TITLE
Fix PIT flaky tests

### PR DESCRIPTION
**Pull request by bharath-techie**
Signed-off-by: Bharathwaj G <bharath78910@gmail.com>

### Description
Fixing couple of flaky tests

PITIt - Sometimes the code to verify no PITs gets executed before deletion of all PITs occur.

PITMultiNodeTests - The node drop message seems to not contain 'partial shards failure' in some cases - so removed the line since assertion anyways contains 'create_pit phase failed' which is good enough for verifying the test behavior.

### Issues Resolved
[List any issues this PR will resolve]

### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
- [ ] Commits are signed per the DCO using --signoff
- [ ] Commit changes are listed out in CHANGELOG.md file (See: [Changelog](../blob/main/CONTRIBUTING.md#changelog))

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -21,6 +21,7 @@ Inspired from [Keep a Changelog](https://keepachangelog.com/en/1.0.0/)
 - [Segment Replication] Add snapshot and restore tests for segment replication feature ([#3993](https://github.com/opensearch-project/OpenSearch/pull/3993))
 - Added missing javadocs for `:example-plugins` modules ([#4540](https://github.com/opensearch-project/OpenSearch/pull/4540))
 - Add missing Javadoc tag descriptions in unit tests ([#4629](https://github.com/opensearch-project/OpenSearch/pull/4629))
+- Fixing PIT flaky tests ([4632](https://github.com/opensearch-project/OpenSearch/pull/4632))
```
Comment (pranikum) [inline CHANGELOG.md:24]: Should we add this in Fixed section?

**Review by pranikum — APPROVED** [review]

**Comment by sachinpkale** [comment]
In the description, please explain what makes the tests flaky and what changes are being made to fix the flakiness.

**Comment by sachinpkale** [comment]
Use imperative mood while writing PR titles.

**Comment by sachinpkale** [comment]
Thanks for fixing these tests.

**Comment by bharath-techie** [comment]
@reta @Bukhtawar can you please review and merge ?

File: client/rest-high-level/src/test/java/org/opensearch/client/PitIT.java [hunk 2]
```diff
@@ -123,6 +127,7 @@ public void onFailure(Exception e) {
         };
         highLevelClient().getAllPitsAsync(RequestOptions.DEFAULT, getPitsListener);
         highLevelClient().deleteAllPitsAsync(RequestOptions.DEFAULT, deletePitListener);
+        countDownLatch.await(10, TimeUnit.SECONDS);
```
Comment (Bukhtawar) [inline client/rest-high-level/src/test/java/org/opensearch/client/PitIT.java:130]: `assertTrue(countDownLatch.await(10, TimeUnit.SECONDS);'

Comment (Bukhtawar) [inline client/rest-high-level/src/test/java/org/opensearch/client/PitIT.java:130]:
```suggestion
        assertTrue(countDownLatch.await(10, TimeUnit.SECONDS);
```

**Review by Bukhtawar — APPROVED** [review]

**Pull request merged**